aequilibrae/paths/route_choice.py

```diff
@@ -0,0 +1,293 @@
+        """
+        if len(nodes) == 0:
+            raise ValueError("`nodes` list-like empty.")
+
+        if isinstance(nodes[0], tuple):
+            # Selection of OD pairs
+            if any(len(x) != 2 for x in nodes):
+                raise ValueError("`nodes` list contains non-pair elements")
+            self.nodes = nodes
+
+        elif isinstance(nodes[0], int):
+            self.nodes = list(itertools.permutations(nodes, r=2))
+
+    def execute_single(self, origin: int, destination: int):
+        if self.__rc is None:
+            self.__rc = RouteChoiceSet(self.graph)
+
+        return self.__rc.run(origin, destination, **self.paramaters)
+
+    def execute(self, path_size_logit: bool = False):
+        if self.__rc is None:
+            self.__rc = RouteChoiceSet(self.graph)
+
+        return self.__rc.batched(
+            self.nodes, bfsle=self.algorithm == "bfsle", path_size_logit=path_size_logit, **self.paramaters
+        )
+
+    def info(self) -> dict:
+        """Returns information for the transit assignment procedure
+
+        Dictionary contains keys  'Algorithm', 'Matrix totals', 'Computer name', 'Procedure ID'.
+
+        The classes key is also a dictionary with all the user classes per transit class and their respective
+        matrix totals
+
+        :Returns:
+            **info** (:obj:`dict`): Dictionary with summary information
+        """
+
+        matrix_totals = {nm: np.sum(self.matrix.matrix_view[:, :, i]) for i, nm in enumerate(self.matrix.view_names)}
+
+        info = {
+            "Algorithm": self.algorithm,
+            "Matrix totals": matrix_totals,
+            "Computer name": socket.gethostname(),
+            "Procedure ID": self.procedure_id,
+            "Parameters": self.paramaters,
+        }
+        return info
+
+    def log_specification(self):
+        self.logger.info("Route Choice specification")
+        self.logger.info(self._config)
+
+    def results(self):
+        """Returns the results of the route choice procedure
+
+        Returns a table of OD pairs to lists of link IDs for each OD pair provided (as columns). Represents paths from ``origin`` to ``destination``.
+
+        :Returns:
+            **results** (:obj:`pa.Table`): Table with the results of the route choice procedure
+
+        """
+        if self.results is None:
+            try:
+                self.results = self.__rc.get_results()
+            except RuntimeError as err:
+                if self.where is None:
+                    raise ValueError("Route choice results not computed and read/save path not specificed") from err
+                self.results = pa.dataset.dataset(
+                    self.where, format="parquet", partitioning=pa.dataset.HivePartitioning(self.schema)
+                )
+
+        return self.results
+
+    def get_load_results(
+        self, which: str = "uncompressed"
+    ) -> Union[Tuple[AequilibraeData, AequilibraeData], Tuple[AequilibraeData]]:
```

Comment:
Let's replace AequilibraEData with PandasDataFrame.  I want to start deprecating AequilibraEData in favor of DFs and Parquet/Feather